SimG4CMS/HGCalTestBeam/interface/AHCalDetId.h

```diff
@@ -0,0 +1,55 @@
+#ifndef SimG4CMS_HGCalTestBeam_HCALDETID_H
+#define SimG4CMS_HGCalTestBeam_HCALDETID_H 1
+
+#include <iosfwd>
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/HcalDetId/interface/HcalSubdetector.h"
+#include "DataFormats/HcalDetId/interface/HcalDetId.h"
+
+
+/** \class AHCalDetId
+ *  Cell identifier class for the HCAL subdetectors, precision readout cells only
+ */
+class AHCalDetId : public DetId {
+public:
+  /** Create a null cellid*/
+  AHCalDetId();
+  /** Create cellid from raw id (0=invalid tower id) */
+  AHCalDetId(uint32_t rawid);
+  /** Constructor from subdetector, signed tower ieta,iphi,and depth */
+  AHCalDetId(int row, int col, int depth);
+  /** Constructor from a generic cell id */
+  AHCalDetId(const DetId& id);
+  /** Assignment from a generic cell id */
+  AHCalDetId& operator=(const DetId& id) {
+    id_=id.rawId();
+  return *this;
+  }
+
+  /// get the subdetector
+  HcalSubdetector subdet() const { return HcalOther; }
+  /// get the z-side of the cell (1/-1)
+  int zside() const { return 1; }
```

Comment:
Maybe better to have this return `HcalDetId(id_).zside()`; then you could use `AHCalDetId` consistently on `HGCalTBAnalyzer.cc:549`.